### ISSUE TITLE
Fatal error: Cannot read property 'rawRequest' of undefined

### ISSUE THREAD

**Issue by grrowl** [Turbo87/webpack-notifier#5]
```
Fatal error: Cannot read property 'rawRequest' of undefined
TypeError: Cannot read property 'rawRequest' of undefined
    at WebpackNotifierPlugin.compileMessage (/Users/tom/Documents/magicland/auth.comet.is/node_modules/webpack-notifier/index.js:29:31)
    at WebpackNotifierPlugin.compilationDone (/Users/tom/Documents/magicland/auth.comet.is/node_modules/webpack-notifier/index.js:39:20)
    at Tapable.applyPlugins (/Users/tom/Documents/magicland/auth.comet.is/node_modules/webpack/node_modules/tapable/lib/Tapable.js:26:37)
    at Tapable.<anonymous> (/Users/tom/Documents/magicland/auth.comet.is/node_modules/webpack/lib/Compiler.js:191:12)
    at FSReqWrap.oncomplete (fs.js:99:15)
```

Added simply with `plugins: [ new WebpackNotifierPlugin({title: 'server build'}) ]`

**Comment by Turbo87** [comment]
looks like `error.module` is undefined. can you reproduce that error?

**Comment by grrowl** [comment]
It took me a second, but yes! I'm building two entry points concurrently, with something similar to the below Gruntfile setup. I don't think `grunt-concurrent` does "real" process forking to achieve this so it might be clobbering a static variable somewhere within webpack (unorthodox, i know):

```js
module.exports = function (grunt) {

  var webpack = require('webpack'),
      webpackConfig = require('./webpack.config.js'); // returns global config plus env-specific tweaks

  // Load all available grunt- npm tasks
  require('matchdep').filterAll('grunt-*').forEach(grunt.loadNpmTasks);

  grunt.initConfig({
    webpack: {
      options: {},
      'client-prod': webpackConfig('client'),
      'server-prod': webpackConfig('server')
    },

    // Tasks for compiling the app concurrently (output may be a bit messy)
    concurrent: {
      'compile-prod': {
        tasks: ['webpack:server-prod', 'webpack:client-prod'],
        options: {logConcurrentOutput: true}
      }
    }
  });

  grunt.registerTask('build:prod', function () {
    grunt.task.run([
      'concurrent:compile-prod',
    ]);
  });

};
```

**Comment by grrowl** [comment]
The above cut-down config actually works as `grunt build:prod`, so it must be one of the removed tasks. I'll dig into it tomorrow and follow-up.

**Comment by Turbo87** [comment]
@grrowl it would be awesome if you could modify https://github.com/Turbo87/webpack-notifier/tree/master/example to reproduce this.

**Comment by DelvarWorld** [comment]
I also hit this on a very large build. Copying from https://github.com/webpack/webpack/issues/930

full stack trace:

    $ webpack -p

        ./node_modules/webpack-notifier/index.js:29
            var message = error.module.rawRequest;
                                    ^
        TypeError: Cannot read property 'rawRequest' of undefined
            at WebpackNotifierPlugin.compileMessage (./node_modules/webpack-notifier/index.js:29:31)
            at WebpackNotifierPlugin.compilationDone (./node_modules/webpack-notifier/index.js:39:20)
            at Tapable.applyPlugins (./node_modules/webpack/node_modules/tapable/lib/Tapable.js:26:37)
            at Tapable.<anonymous> (./node_modules/webpack/lib/Compiler.js:189:12)
            at Tapable.emitRecords (./node_modules/webpack/lib/Compiler.js:277:37)
            at Tapable.<anonymous> (./node_modules/webpack/lib/Compiler.js:183:11)
            at ./node_modules/webpack/lib/Compiler.js:270:11
            at Tapable.applyPluginsAsync (./node_modules/webpack/node_modules/tapable/lib/Tapable.js:60:69)
            at Tapable.afterEmit (./node_modules/webpack/lib/Compiler.js:267:8)
            at Tapable.<anonymous> (./node_modules/webpack/lib/Compiler.js:262:14)

Plugins list:


    plugins: [
        new webpack.ProvidePlugin({
            $: 'jquery',
            jQuery: 'jquery'
        }),
        new WebpackNotifierPlugin(),
        new webpack.optimize.UglifyJsPlugin({
            compress: {
                warnings: false
            }
        })
    ]

When I remove the notifier, I get a very large amount of warnings from uglify, which might be related?

    $ webpack -p

        Hash: 8b99c30c6cfe657f0be5
        Version: webpack 1.6.0
        Time: 49426ms
            Asset     Size  Chunks             Chunk Names
        bundle.js  1419317       0  [emitted]  main
            + 330 hidden modules

        WARNING in bundle.js from UglifyJs

(616 warnings follow)

**Comment by zachwood** [comment]
Same issue as @DelvarWorld, if I have Uglify warnings I get the error:

    TypeError: Cannot read property 'rawRequest' of undefined

If uglification is skipped (and thus no warning output), everything works as expected.

**Comment by DelvarWorld** [comment]
FYI, the workaround I did this is that for prod builds, I run the webpack task with `NODE_ENV=production` and in webpack.config.js build the plugin list dynamically based on if `process.env.NODE_ENV === 'production'` . If it's true I add Uglify plugin, otherwise I add Notifier plugin. You probably shouldn't be uglifying in dev and you shouldn't be notifying in production, so this works well for me.

**Comment by grrowl** [comment]
Sorry I never followed up on this — been swamped and have moved away from including this for now (other devs found it noisy)

**Comment by Turbo87** [comment]
I just pushed a fix for this issue and released it as v1.2.1. Please check if this issue is resolved for you now and let me know if there are still problems.

**Comment by zachwood** [comment]
Working great for me. Thanks!
